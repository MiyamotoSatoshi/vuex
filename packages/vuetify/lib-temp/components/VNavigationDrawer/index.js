import VNavigationDrawer from './VNavigationDrawer';
export { VNavigationDrawer };
export default VNavigationDrawer;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTmF2aWdhdGlvbkRyYXdlci9pbmRleC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLGlCQUFpQixNQUFNLHFCQUFxQixDQUFBO0FBRW5ELE9BQU8sRUFBRSxpQkFBaUIsRUFBRSxDQUFBO0FBQzVCLGVBQWUsaUJBQWlCLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgVk5hdmlnYXRpb25EcmF3ZXIgZnJvbSAnLi9WTmF2aWdhdGlvbkRyYXdlcidcblxuZXhwb3J0IHsgVk5hdmlnYXRpb25EcmF3ZXIgfVxuZXhwb3J0IGRlZmF1bHQgVk5hdmlnYXRpb25EcmF3ZXJcbiJdfQ==