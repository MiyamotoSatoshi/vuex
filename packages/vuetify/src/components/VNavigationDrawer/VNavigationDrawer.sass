@import '../../styles/styles.sass'

// Theme
+theme(v-navigation-drawer) using ($material)
  background-color: map-get($material, 'cards')

  &:not(.v-navigation-drawer--floating)
    .v-navigation-drawer__border
      background-color: map-get($material, 'dividers')

  .v-divider
    border-color: map-get($material, 'dividers')

// Block
.v-navigation-drawer
  -webkit-overflow-scrolling: touch
  display: flex
  flex-direction: column
  left: 0
  max-width: 100%
  overflow: hidden
  pointer-events: auto
  top: 0
  transition-duration: 0.2s
  transition-timing-function: map-get($transition, 'fast-out-slow-in')
  will-change: transform
  transition-property: transform, width
  +bootable()

  &.v-navigation-drawer--right
    &:after
      left: 0
      right: initial

  .v-list
    background: inherit

// Element
.v-navigation-drawer__border
    position: absolute
    right: 0
    top: 0
    height: 100%
    width: 1px

.v-navigation-drawer__content
  height: 100%
  overflow-y: auto
  overflow-x: hidden

.v-navigation-drawer__image
  border-radius: inherit
  height: 100%
  position: absolute
  top: 0
  bottom: 0
  z-index: -1
  contain: strict
  width: 100%

  .v-image
    border-radius: inherit

// Modifier
.v-navigation-drawer--bottom.v-navigation-drawer--is-mobile
  max-height: 50%
  top: auto
  bottom: 0
  min-width: 100%

.v-navigation-drawer--right
  left: auto
  right: 0

  > .v-navigation-drawer__border
    right: auto
    left: 0

.v-navigation-drawer--absolute
  z-index: 1

.v-navigation-drawer--fixed
  z-index: 6

.v-navigation-drawer--absolute
  position: absolute

.v-navigation-drawer--clipped
  z-index: 4

.v-navigation-drawer--fixed
  position: fixed

.v-navigation-drawer--floating:after
  display: none

.v-navigation-drawer--mini-variant
  overflow: hidden

  .v-list-item
    justify-content: center

    > *:first-child
      margin-left: 0
      margin-right: 0

    > *:not(:first-child)
      display: none

  .v-list-group--no-action .v-list-group__items,
  .v-list-group--sub-group
    display: none

.v-navigation-drawer--temporary,
.v-navigation-drawer--is-mobile
  z-index: 6

  &:not(.v-navigation-drawer--close)
    +elevation(16)
