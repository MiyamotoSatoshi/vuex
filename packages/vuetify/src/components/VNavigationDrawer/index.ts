export { default as VNavigationDrawer } from './VNavigationDrawer'
