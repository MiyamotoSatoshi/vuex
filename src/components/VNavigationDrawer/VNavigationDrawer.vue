<script>
  import Overlayable from '../../mixins/overlayable'
  import Themeable from '../../mixins/themeable'

  import ClickOutside from '../../directives/click-outside'
  import Resize from '../../directives/resize'
  import Touch from '../../directives/touch'

  export default {
    name: 'v-navigation-drawer',

    mixins: [Overlayable, Themeable],

    directives: {
      ClickOutside,
      Resize,
      Touch
    },

    data () {
      return {
        isActive: this.value,
        isBooted: false,
        isMobile: false,
        touchArea: {
          left: 0,
          right: 0
        }
      }
    },

    props: {
      absolute: Boolean,
      clipped: Boolean,
      disableRouteWatcher: Boolean,
      enableResizeWatcher: Boolean,
      height: String,
      floating: Boolean,
      miniVariant: Boolean,
      mobileBreakPoint: {
        type: Number,
        default: 1280
      },
      permanent: Boolean,
      persistent: Boolean,
      right: Boolean,
      temporary: Boolean,
      touchless: Boolean,
      value: { required: false }
    },

    computed: {
      calculatedHeight () {
        return this.height || '100%'
      },
      classes () {
        return {
          'navigation-drawer': true,
          'navigation-drawer--absolute': this.absolute,
          'navigation-drawer--clipped': this.clipped,
          'navigation-drawer--close': !this.isActive,
          'navigation-drawer--floating': this.floating,
          'navigation-drawer--is-booted': this.isBooted,
          'navigation-drawer--is-mobile': this.isMobile,
          'navigation-drawer--mini-variant': this.miniVariant,
          'navigation-drawer--open': this.isActive,
          'navigation-drawer--permanent': this.permanent,
          'navigation-drawer--persistent': this.persistent,
          'navigation-drawer--right': this.right,
          'navigation-drawer--temporary': this.temporary,
          'theme--dark': this.dark,
          'theme--light': this.light
        }
      },
      showOverlay () {
        return !this.permanent &&
          this.isActive &&
          (this.temporary || this.isMobile)
      }
    },

    watch: {
      $route () {
        if (!this.disableRouteWatcher) {
          this.isActive = !this.closeConditional()
        }
      },
      isActive (val) {
        this.$emit('input', val)
        this.showOverlay &&
          val &&
          this.genOverlay() ||
          this.removeOverlay()
        this.$el.scrollTop = 0
      },
      isMobile (val) {
        !val && this.removeOverlay()
      },
      permanent (val) {
        this.$emit('input', val)
      },
      value (val) {
        if (this.permanent) return
        if (val !== this.isActive) this.isActive = val
      }
    },

    mounted () {
      this.$vuetify.load(this.init)
    },

    methods: {
      init () {
        this.checkIfMobile()
        setTimeout(() => (this.isBooted = true), 0)

        if (this.permanent) this.isActive = true
        else if (this.isMobile) this.isActive = false
        else if (!this.value &&
          (this.persistent || this.temporary)
        ) this.isActive = false
        else this.isActive = true
      },
      checkIfMobile () {
        this.isMobile = window.innerWidth < parseInt(this.mobileBreakPoint)
      },
      closeConditional () {
        return !this.permanent && (this.temporary || this.isMobile)
      },
      onResize () {
        if (!this.enableResizeWatcher ||
          this.permanent ||
          this.temporary
        ) return

        this.checkIfMobile()
        this.isActive = !this.isMobile
      },
      swipeRight (e) {
        if (this.isActive && !this.right) return
        this.calculateTouchArea()

        if (Math.abs(e.touchendX - e.touchstartX) < 100) return
        else if (!this.right &&
          e.touchstartX <= this.touchArea.left
        ) this.isActive = true
        else if (this.right && this.isActive) this.isActive = false
      },
      swipeLeft (e) {
        if (this.isActive && this.right) return
        this.calculateTouchArea()

        if (Math.abs(e.touchendX - e.touchstartX) < 100) return
        else if (this.right &&
          e.touchstartX >= this.touchArea.right
        ) this.isActive = true
        else if (!this.right && this.isActive) this.isActive = false
      },
      calculateTouchArea () {
        if (!this.$el.parentNode) return
        const parentRect = this.$el.parentNode.getBoundingClientRect()

        this.touchArea = {
          left: parentRect.left + 50,
          right: parentRect.right - 50
        }
      },
      genDirectives () {
        const directives = [
          {
            name: 'click-outside',
            value: this.closeConditional
          },
          {
            name: 'resize',
            value: this.onResize
          }
        ]

        !this.touchless && directives.push({
          name: 'touch',
          value: {
            parent: true,
            left: this.swipeLeft,
            right: this.swipeRight
          }
        })

        return directives
      }
    },

    render (h) {
      const data = {
        'class': this.classes,
        style: { height: this.calculatedHeight },
        directives: this.genDirectives(),
        on: {
          click: () => this.$emit('update:miniVariant', false)
        }
      }

      return h('aside', data, [
        this.$slots.default,
        h('div', { 'class': 'navigation-drawer__border' })
      ])
    }
  }
</script>

<style lang="stylus" src="../../stylus/components/_navigation-drawer.styl"></style>
