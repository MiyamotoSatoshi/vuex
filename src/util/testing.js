import Vue from 'vue'
import load from '~util/load'
import { mount, shallow } from 'avoriaz'

export function test(name, cb) {
  Vue.prototype.$vuetify = { load }

  beforeEach(() => {
    rafPolyfill(window)
  })

  describe(name, () => cb({
    functionalContext,
    mount,
    shallow
  }))
}

export function functionalContext(context = {}) {
  return {
    context: Object.assign({
      data: {},
      props: {}
    }, context)
  }
}


//requestAnimationFrame polyfill | Milos Djakonovic ( @Miloshio ) | MIT | https://github.com/milosdjakonovic/requestAnimationFrame-polyfill
export function rafPolyfill(w) {
  /**
   *
   * How many times should polyfill call
   * update callback? By canon, it should
   * be 60 times per second, so that ideal
   * framerate 60fps could be reached.
   *
   * However, even native implementations
   * of requestAnimationFrame often cannot
   * do 60fps, but, unlike any polyfill,
   * they can synchronise achievable fps
   * rate with screen refresh rate.
   *
   * So, leave this value 1000/60 unless
   * you target specific browser on spec
   * ific device that is going to work
   * better with custom value. I think
   * that this is the longest comment I've
   * written on single variable so far.
  **/
  var FRAME_RATE_INTERVAL = 1000/60,

  /**
   * All queued callbacks in given cycle.
  **/
  allCallbacks = [],

  executeAllScheduled = false,

  shouldCheckCancelRaf = false,

  /**
   * Callbacks queued for cancellation.
  **/
  callbacksForCancellation = [],

  /**
   * Should callback be cancelled?
   * @param cb - callback
  **/
  isToBeCancelled = function(cb){
    for(var i=0;i<callbacksForCancellation.length;i++){
      if(callbacksForCancellation[i] === cb ){
        callbacksForCancellation.splice(i,1);
        return true;
      }
    }
  },



  /**
   *
   * Executes all (surprise) callbacks in
   * and removes them from callback queue.
   *
  **/
  executeAll = function(){
    executeAllScheduled = false;
    var _allCallbacks = allCallbacks;
    allCallbacks = [];
    for(var i=0;i<_allCallbacks.length;i++){
      if(shouldCheckCancelRaf===true){
        if (isToBeCancelled(_allCallbacks[i])){
          shouldCheckCancelRaf = false;
          return;
        }
      }
      _allCallbacks[i].apply(w, [ new Date().getTime() ] );
    }
  },

  /**
   *
   * requestAnimationFrame polyfill
   * @param callback - callback to be queued & executed | executed
   * @return callback
   *
  **/
  raf = function(callback){
    allCallbacks.push(callback);
    if(executeAllScheduled===false){
      w.setTimeout(executeAll, FRAME_RATE_INTERVAL);
      executeAllScheduled = true;
    }
    return callback;
  },

  /**
   *
   * Cancels raf.
  **/
  cancelRaf = function(callback){
    callbacksForCancellation.push(callback);
    shouldCheckCancelRaf = true;
  },


  //https://gist.github.com/paulirish/1579671
  vendors = ['ms', 'moz', 'webkit', 'o'];

    for(var x = 0; x < vendors.length && !w.requestAnimationFrame; ++x) {
        w.requestAnimationFrame = w[vendors[x]+'RequestAnimationFrame'];
        w.cancelAnimationFrame = w[vendors[x]+'CancelAnimationFrame']
        || w[vendors[x]+'CancelRequestAnimationFrame'];
    }

  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf;
  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;

}
