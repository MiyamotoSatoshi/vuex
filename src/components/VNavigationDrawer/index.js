import VNavigationDrawer from './VNavigationDrawer'

/* istanbul ignore next */
VNavigationDrawer.install = function install (Vue) {
  Vue.component(VNavigationDrawer.name, VNavigationDrawer)
}

export default VNavigationDrawer
