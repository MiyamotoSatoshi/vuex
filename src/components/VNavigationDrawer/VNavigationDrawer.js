import '../../stylus/components/_navigation-drawer.styl'

// Mixins
import Applicationable from '../../mixins/applicationable'
import Overlayable from '../../mixins/overlayable'
import SSRBootable from '../../mixins/ssr-bootable'
import Themeable from '../../mixins/themeable'

// Directives
import ClickOutside from '../../directives/click-outside'
import Resize from '../../directives/resize'
import Touch from '../../directives/touch'

// Helpers
import { convertToUnit } from '../../util/helpers'

/* @vue/component */
export default {
  name: 'v-navigation-drawer',

  directives: {
    ClickOutside,
    Resize,
    Touch
  },

  mixins: [
    Applicationable(null, [
      'miniVariant',
      'right',
      'width'
    ]),
    Overlayable,
    SSRBootable,
    Themeable
  ],

  props: {
    clipped: Boolean,
    disableRouteWatcher: Boolean,
    disableResizeWatcher: Boolean,
    height: {
      type: [Number, String],
      default: '100%'
    },
    floating: Boolean,
    miniVariant: Boolean,
    miniVariantWidth: {
      type: [Number, String],
      default: 80
    },
    mobileBreakPoint: {
      type: [Number, String],
      default: 1264
    },
    permanent: Boolean,
    right: Boolean,
    stateless: Boolean,
    temporary: Boolean,
    touchless: Boolean,
    width: {
      type: [Number, String],
      default: 300
    },
    value: { required: false }
  },

  data: () => ({
    isActive: false,
    touchArea: {
      left: 0,
      right: 0
    }
  }),

  computed: {
    /**
     * Used for setting an app
     * value from a dynamic
     * property. Called from
     * applicationable.js
     *
     * @return {string}
     */
    applicationProperty () {
      return this.right ? 'right' : 'left'
    },
    calculatedTransform () {
      if (this.isActive) return 0

      return this.right
        ? this.calculatedWidth
        : -this.calculatedWidth
    },
    calculatedWidth () {
      return this.miniVariant
        ? this.miniVariantWidth
        : this.width
    },
    classes () {
      return {
        'v-navigation-drawer': true,
        'v-navigation-drawer--absolute': this.absolute,
        'v-navigation-drawer--clipped': this.clipped,
        'v-navigation-drawer--close': !this.isActive,
        'v-navigation-drawer--fixed': !this.absolute && (this.app || this.fixed),
        'v-navigation-drawer--floating': this.floating,
        'v-navigation-drawer--is-mobile': this.isMobile,
        'v-navigation-drawer--mini-variant': this.miniVariant,
        'v-navigation-drawer--open': this.isActive,
        'v-navigation-drawer--right': this.right,
        'v-navigation-drawer--temporary': this.temporary,
        'theme--dark': this.dark,
        'theme--light': this.light
      }
    },
    hasApp () {
      return this.app &&
        (!this.isMobile && !this.temporary)
    },
    isMobile () {
      return !this.stateless &&
        !this.permanent &&
        !this.temporary &&
        this.$vuetify.breakpoint.width < parseInt(this.mobileBreakPoint, 10)
    },
    marginTop () {
      if (!this.hasApp) return 0

      let marginTop = this.$vuetify.application.bar

      marginTop += this.clipped
        ? this.$vuetify.application.top
        : 0

      return marginTop
    },
    maxHeight () {
      if (!this.hasApp) return null

      const maxHeight = (
        this.$vuetify.application.bottom +
        this.$vuetify.application.footer +
        this.$vuetify.application.bar
      )

      if (!this.clipped) return maxHeight

      return maxHeight + this.$vuetify.application.top
    },
    reactsToClick () {
      return !this.stateless &&
        !this.permanent &&
        (this.isMobile || this.temporary)
    },
    reactsToMobile () {
      return !this.disableResizeWatcher &&
        !this.stateless &&
        !this.permanent &&
        !this.temporary
    },
    reactsToRoute () {
      return !this.disableRouteWatcher &&
        !this.stateless &&
        (this.temporary || this.isMobile)
    },
    resizeIsDisabled () {
      return this.disableResizeWatcher || this.stateless
    },
    showOverlay () {
      return this.isActive &&
        (this.isMobile || this.temporary)
    },
    styles () {
      const styles = {
        height: convertToUnit(this.height),
        marginTop: `${this.marginTop}px`,
        maxHeight: `calc(100% - ${+this.maxHeight}px)`,
        transform: `translateX(${this.calculatedTransform}px)`,
        width: `${this.calculatedWidth}px`
      }

      return styles
    }
  },

  watch: {
    $route () {
      if (this.reactsToRoute && this.closeConditional()) {
        this.isActive = false
      }
    },
    isActive (val) {
      this.$emit('input', val)
      this.callUpdate()
    },
    /**
     * When mobile changes, adjust
     * the active state only when
     * there has been a previous
     * value
     */
    isMobile (val, prev) {
      !val &&
        this.isActive &&
        !this.temporary &&
        this.removeOverlay()

      if (prev == null ||
        this.resizeIsDisabled ||
        !this.reactsToMobile
      ) return

      this.isActive = !val
      this.callUpdate()
    },
    permanent (val) {
      // If enabling prop
      // enable the drawer
      if (val) {
        this.isActive = true
      }
      this.callUpdate()
    },
    showOverlay (val) {
      if (val) this.genOverlay()
      else this.removeOverlay()
    },
    temporary () {
      this.callUpdate()
    },
    value (val) {
      if (this.permanent) return

      if (val == null) return this.init()

      if (val !== this.isActive) this.isActive = val
    }
  },

  beforeMount () {
    this.init()
  },

  methods: {
    calculateTouchArea () {
      if (!this.$el.parentNode) return
      const parentRect = this.$el.parentNode.getBoundingClientRect()

      this.touchArea = {
        left: parentRect.left + 50,
        right: parentRect.right - 50
      }
    },
    closeConditional () {
      return this.isActive && this.reactsToClick
    },
    genDirectives () {
      const directives = [{
        name: 'click-outside',
        value: () => (this.isActive = false),
        args: {
          closeConditional: this.closeConditional
        }
      }]

      !this.touchless && directives.push({
        name: 'touch',
        value: {
          parent: true,
          left: this.swipeLeft,
          right: this.swipeRight
        }
      })

      return directives
    },
    /**
     * Sets state before mount to avoid
     * entry transitions in SSR
     *
     * @return {void}
     */
    init () {
      if (this.permanent) {
        this.isActive = true
      } else if (this.stateless ||
        this.value != null
      ) {
        this.isActive = this.value
      } else if (!this.temporary) {
        this.isActive = !this.isMobile
      }
    },
    swipeRight (e) {
      if (this.isActive && !this.right) return
      this.calculateTouchArea()

      if (Math.abs(e.touchendX - e.touchstartX) < 100) return
      if (!this.right &&
        e.touchstartX <= this.touchArea.left
      ) this.isActive = true
      else if (this.right && this.isActive) this.isActive = false
    },
    swipeLeft (e) {
      if (this.isActive && this.right) return
      this.calculateTouchArea()

      if (Math.abs(e.touchendX - e.touchstartX) < 100) return
      if (this.right &&
        e.touchstartX >= this.touchArea.right
      ) this.isActive = true
      else if (!this.right && this.isActive) this.isActive = false
    },
    /**
     * Update the application layout
     *
     * @return {number}
     */
    updateApplication () {
      return !this.isActive ||
        this.temporary ||
        this.isMobile
        ? 0
        : this.calculatedWidth
    }
  },

  render (h) {
    const data = {
      'class': this.classes,
      style: this.styles,
      directives: this.genDirectives(),
      on: {
        click: () => {
          if (!this.miniVariant) return

          this.$emit('update:miniVariant', false)
        },
        transitionend: e => {
          if (e.target !== e.currentTarget) return
          this.$emit('transitionend', e)

          // IE11 does not support new Event('resize')
          const resizeEvent = document.createEvent('UIEvents')
          resizeEvent.initUIEvent('resize', true, false, window, 0)
          window.dispatchEvent(resizeEvent)
        }
      }
    }

    return h('aside', data, [
      this.$slots.default,
      h('div', { 'class': 'v-navigation-drawer__border' })
    ])
  }
}
