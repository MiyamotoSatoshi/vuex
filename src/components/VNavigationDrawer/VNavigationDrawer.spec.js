import VNavigationDrawer from '~components/VNavigationDrawer'
import { test } from '~util/testing'

// TODO: Test behaviour instead of styles
test('VNavigationDrawer.js', ({ mount }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom absolute and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        absolute: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom clipped and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        clipped: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom disableRouteWatcher and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        disableRouteWatcher: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom enableResizeWatcher and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        enableResizeWatcher: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom height and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        height: '100px'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom floating and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        floating: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom miniVariant and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        miniVariant: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom mobileBreakPoint and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        mobileBreakPoint: 1000
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom permanent and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        permanent: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom persistent and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        persistent: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom right and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        right: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom temporary and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        temporary: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom touchless and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        touchless: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with custom value and match snapshot', () => {
    const wrapper = mount(VNavigationDrawer, {
      propsData: {
        value: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match value if resize-watcher is not enabled', async () => {
    const wrapper = mount(VNavigationDrawer, {
      attachToDocument: true,
      propsData: {
        permanent: true,
        value: false
      }
    })

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.isActive).toBe(false)
  })
})
