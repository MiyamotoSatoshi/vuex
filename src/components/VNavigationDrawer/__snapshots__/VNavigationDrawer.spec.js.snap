// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VNavigationDrawer.js should render component and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom absolute and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--absolute navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom clipped and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--clipped navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom disableRouteWatcher and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom enableResizeWatcher and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom floating and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--floating"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom height and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100px; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom miniVariant and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--mini-variant"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 80px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom mobileBreakPoint and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom permanent and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--permanent"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom persistent and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--persistent"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom right and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--right"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom temporary and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close navigation-drawer--temporary"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom touchless and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--close"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;

exports[`VNavigationDrawer.js should render component with custom value and match snapshot 1`] = `

<aside class="navigation-drawer navigation-drawer--open"
       style="height: 100%; margin-top: 0px; max-height: calc(100% - 100%px); width: 300px;"
>
  <div class="navigation-drawer__border">
  </div>
</aside>

`;
